src/sentry/search/snuba/backend.py

```diff
@@ -225,20 +226,10 @@ def assigned_or_suggested_filter(
     if "User" in types_to_owners:
         users = types_to_owners["User"]
         user_ids: List[int] = [u.id for u in users if u is not None]
-        team_ids = list(
-            Team.objects.filter(
-                id__in=OrganizationMemberTeam.objects.filter(
-                    organizationmember__in=OrganizationMember.objects.filter(
-                        user_id__in=user_ids, organization_id=organization_id
-                    ),
-                    is_active=True,
-                ).values("team")
-            ).values_list("id", flat=True)
-        )
```

Comment:
I think we should keep this PR small and leave changes to `assigned_or_suggested` filter to another PR. What do you think?

Reply:
I brought that code back and put it under the feature flag. Removing the team_id from the query is necessary for the changes to work. Does that address your question?

Reply:
Maybe I'm missing something, but I'm not sure changes to `assigned_or_suggested` are needed to get `assigned_to` filter to work? It might make sense to break up the PR so this PR is only changing code for `assigned_to/assigned` related functionality and open another PR for `assigned_or_suggested` changes. 

This has the benefit of keeping the scope of your PRs smaller and more manageable for reviewers to grok.

Reply:
I see what you mean. I have moved the logic to a new PR: https://github.com/getsentry/sentry/pull/51200